cpu/stm32/include/periph/cpu_i2c.h

```diff
@@ -40,37 +51,43 @@ extern "C" {
 #define PERIPH_I2C_NEED_WRITE_REG
 /** Use read regs function from periph common */
 #define PERIPH_I2C_NEED_READ_REGS
-#if defined(CPU_FAM_STM32F1) || defined(CPU_FAM_STM32F2) || \
-    defined(CPU_FAM_STM32L1) || defined(CPU_FAM_STM32F4) || \
-    defined(CPU_FAM_STM32MP1)
+#if STM32_I2C_VARIANT == 2
```

Comment:
`STM32_I2C_VARIANT` is defined a few lines below this.

Reply:
Fixed